tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Linker.scala

```diff
@@ -243,6 +270,43 @@ final class Linker(semantics: Semantics, outputMode: OutputMode,
         version)
   }
 
+  private def synthesizeInheritedConstructor(
+      classInfo: Analysis.ClassInfo, methodInfo: Analysis.MethodInfo,
+      getTree: TreeProvider, analysis: Analysis)(
+      implicit pos: Position): MethodDef = {
+    val encodedName = methodInfo.encodedName
```

Comment:
Don't forget overloads. `init___` is not the only constructor name. There are plenty of them, and they're definitely not constant.